src/components/dialogs/VerifyEmailDialog.tsx

```diff
@@ -0,0 +1,247 @@
+      }}>
+      <Dialog.Handle />
+      <Inner control={control} setDidVerify={setDidVerify} />
+    </Dialog.Outer>
+  )
+}
+
+export function Inner({
+  control,
+  setDidVerify,
+}: {
+  control: Dialog.DialogControlProps
+  setDidVerify: (value: boolean) => void
+}) {
+  const {_} = useLingui()
+  const {currentAccount} = useSession()
+  const agent = useAgent()
+  const t = useTheme()
+  const {openModal} = useModalControls()
+  const {gtMobile} = useBreakpoints()
+
+  const [currentStep, setCurrentStep] = React.useState<
+    'StepOne' | 'StepTwo' | 'StepThree'
+  >('StepOne')
+  const [confirmationCode, setConfirmationCode] = React.useState('')
+  const [isProcessing, setIsProcessing] = React.useState(false)
+  const [error, setError] = React.useState('')
+
+  const uiStrings = {
+    StepOne: {
+      title: _(msg`Verify Your Email`),
+      message: '',
+    },
+    StepTwo: {
+      title: _(msg`Enter Code`),
+      message: _(
+        msg`An email has been sent! Please enter the confirmation code included in the email below.`,
+      ),
+    },
+    StepThree: {
+      title: _(msg`Success!`),
+      message: _(msg`Thank you! Your email has been successfully verified.`),
+    },
+  }
+
+  const onSendEmail = async () => {
+    setError('')
+    setIsProcessing(true)
+    try {
+      await agent.com.atproto.server.requestEmailConfirmation()
+      setCurrentStep('StepTwo')
+    } catch (e: unknown) {
+      setError(cleanError(e))
+    } finally {
+      setIsProcessing(false)
+    }
+  }
+
+  const onVerifyEmail = async () => {
+    setError('')
+    setIsProcessing(true)
+    try {
+      await agent.com.atproto.server.confirmEmail({
+        email: (currentAccount?.email || '').trim(),
+        token: confirmationCode.trim(),
+      })
+    } catch (e: unknown) {
+      setError(cleanError(String(e)))
+      setIsProcessing(false)
+      return
+    }
+
+    setIsProcessing(false)
+    setDidVerify(true)
+    setCurrentStep('StepThree')
+  }
+
+  return (
+    <Dialog.ScrollableInner label={_(msg`Verify email dialog`)}>
+      <View style={[a.gap_xl]}>
+        <View style={[a.gap_sm]}>
+          <Text style={[a.font_heavy, a.text_2xl]}>
+            {uiStrings[currentStep].title}
+          </Text>
+          {error ? (
+            <View style={[a.rounded_sm, a.overflow_hidden]}>
+              <ErrorMessage message={error} />
+            </View>
+          ) : null}
+          <Text style={[a.text_md, a.leading_snug]}>
+            {currentStep === 'StepOne' ? (
+              <Trans>
+                You'll receive an email at{' '}
+                <Text style={[a.text_md, a.leading_snug, a.font_bold]}>
+                  {currentAccount?.email}
+                </Text>{' '}
+                to verify it's you.{' '}
+                <Text
+                  onPress={() => {
+                    control.close(() => {
+                      openModal({name: 'change-email'})
+                    })
+                  }}
```

Comment:
Can use `InlineLinkText` with a preventDefault and return false to give us underline on hover

Reply:
This expects `to` as a prop :/